source/cell/cell.cpp

```diff
@@ -767,9 +814,115 @@ bool cell::has_format() const
 
 void cell::format(const class format new_format)
 {
+    // Check if format belongs to a different workbook (dangling pointer risk)
+    if (!workbook().owns_format(new_format))
+    {
+        copy_format_from_other_workbook(new_format);
+        return;
+    }
+
+    // Same workbook: direct assignment (original behavior)
     d_->format_ = new_format.d_;
 }
 
+void cell::copy_format_from_other_workbook(const class format &source_format)
+{
+    // Option B: Automatically clone the format to this workbook's stylesheet
+    // Start with a blank format from our workbook
+    auto cloned_format = workbook().create_format(true);
+
+    // Copy alignment if present
+    if (source_format.d_->alignment_id.is_set())
+    {
+        auto src_alignment = source_format.d_->parent->alignments.at(source_format.d_->alignment_id.get());
+        cloned_format.alignment(src_alignment, source_format.d_->alignment_applied);
+    }
+
+    // Copy border if present
+    if (source_format.d_->border_id.is_set())
+    {
+        auto src_border = source_format.d_->parent->borders.at(source_format.d_->border_id.get());
+        cloned_format.border(src_border, source_format.d_->border_applied);
+    }
+
+    // Copy fill if present
+    if (source_format.d_->fill_id.is_set())
+    {
+        auto src_fill = source_format.d_->parent->fills.at(source_format.d_->fill_id.get());
+        cloned_format.fill(src_fill, source_format.d_->fill_applied);
+    }
+
+    // Copy font if present
+    if (source_format.d_->font_id.is_set())
+    {
+        auto src_font = source_format.d_->parent->fonts.at(source_format.d_->font_id.get());
+        cloned_format.font(src_font, source_format.d_->font_applied);
+    }
+
+    // Copy number format if present
+    if (source_format.d_->number_format_id.is_set())
+    {
+        const auto number_format_id = source_format.d_->number_format_id.get();
```

Comment:
Can't we avoid this code duplication by just calling `source_format.number_format()`?